engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/error/PolyglotError.java

```diff
@@ -27,6 +28,10 @@ public Atom wrap(Context c, IOException e) {
     return newInstance(c.getEnvironment().asGuestValue(e));
   }
 
+  public Atom wrap(Context c, UnsupportedMessageException e) {
```

Comment:
I cannot accept doing this. These `PolyglotError` wrappers are bad enough and really we should get rid of them (with the only exception of `AbstractTruffleException`) and replace with proper domain-specific errors (for IOException, DateTimeException etc.). These are a hack here. And now we're doing something even worse by wrapping an internal truffle class, not meant to be seen by any user (even `Value` translates these into `UnsupportedOperationException`).

Reply:
There is [Simplify exception handling (by removing `Polyglot_Exception`)](https://www.pivotaltracker.com/story/show/182844611) task. Meanwhile let's just _panic_:  4d9dfb440b7208